src/main/java/org/jenkinsci/plugins/mesos/MesosApi.java

```diff
@@ -82,4 +104,29 @@ public MesosApi(String masterUrl, String user, String frameworkName)
         .runWith(Sink.head(), materializer)
         .toCompletableFuture();
   }
+
+  private PodSpec buildMesosAgentTask(double cpu, double mem) {
+    RunSpec spec = new RunSpec(
+        convertListToSeq(Arrays.asList(ScalarRequirement.cpus(cpu), ScalarRequirement.memory(mem))),
+        "echo Hello!",
+        convertListToSeq(Collections.emptyList())
+    );
+    String id = UUID.randomUUID().toString();
+    PodSpec podSpec = new PodSpec(
+        new PodId(String.format("jenkins-test-%s", id)),
+        new Goal.Running$(),
+        spec
+    );
+    return podSpec;
+  }
+
+  public void updateState(StateEvent event) {
+    if (event instanceof PodStateEvent) {
+      stateMap.put(((PodStateEvent) event).id(), event);
```

Comment:
since we have not created a MesosSlave at point of enqueue I changed the map structure here, open for thoughts  here though